### PR TITLE
[Feature][Clickhouse-Connector-V2] Add Clickhouse e2e testcase

### LINKED ISSUE #2949: [Feature][Clickhouse-Connector-V2] Add Clickhouse e2e testcase
Subtask of #2789

### PR DESCRIPTION
**Pull request by FWLamb**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/pom.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>seatunnel-spark-connector-v2-e2e</artifactId>
+        <groupId>org.apache.seatunnel</groupId>
+        <version>${revision}</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>connector-clickhouse-spark-e2e</artifactId>
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/pom.xml]: move to [this](https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e)

reference
https://github.com/apache/incubator-seatunnel/pull/2946

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/clickhouse/ClickhouseIT.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                container.getPassword())) {
+            Statement statement = connection.createStatement();
+            String initializeSourceTableSql = "CREATE TABLE default.source_table (" +
+                    "    `name` Nullable(String)," +
+                    "    `age` Nullable(Int32)" +
+                    ")ENGINE = Memory";
+            String initializeSinkTableSql = "CREATE TABLE default.sink_table (" +
+                    "    `name` Nullable(String)," +
+                    "    `age` Nullable(Int32)" +
+                    ")ENGINE = Memory";
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/clickhouse/ClickhouseIT.java]: Test all supported datatypes?

reference
https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e/connector-mongodb-flink-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/mongodb/MongodbIT.java#L155

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/resources/clickhouse/clickhouse_to_clickhouse.conf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+     result_table_name = "source_table"
+    }
+  # If you would like to get more information about how to configure seatunnel and see full list of source plugins,
+  # please go to https://seatunnel.apache.org/docs/connector-v2/source/FakeSource
+}
+
+transform {
+    sql {
+      sql = "select name,age from source_table"
+    }
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/resources/clickhouse/clickhouse_to_clickhouse.conf]: remove?

test all supported fields

**Comment by liugddx** [comment]
Please solve the problem of CI. please look at https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/setup.md

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/clickhouse/ClickhouseIT.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        Container.ExecResult execResult = executeSeaTunnelSparkJob("/clickhouse/fake_to_clickhouse.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+    }
+
+    @Test
+    public void testClickhouseSourceToClickhouseSink() throws IOException, InterruptedException {
+        Container.ExecResult execResult = executeSeaTunnelSparkJob("/clickhouse/clickhouse_to_clickhouse.conf");
+        Assertions.assertEquals(0, execResult.getExitCode());
+        Assertions.assertIterableEquals(generateTestDataset(), queryResult());
+    }
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/clickhouse/ClickhouseIT.java]:
```suggestion
    @Test
    public void testClickhouseSourceToClickhouseSink() throws IOException, InterruptedException {
        Container.ExecResult execResult = executeSeaTunnelSparkJob("/clickhouse/fake_to_clickhouse.conf");
        Assertions.assertEquals(0, execResult.getExitCode());

        Container.ExecResult execResult = executeSeaTunnelSparkJob("/clickhouse/clickhouse_to_clickhouse.conf");
        Assertions.assertEquals(0, execResult.getExitCode());
        Assertions.assertIterableEquals(generateTestDataset(), queryResult());
    }
```
merge together

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/pom.xml (hunk 1 above)
Comment (FWLamb) [reply seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/pom.xml]: > move to [this](https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e)
> 
> reference #2946

I want to know the relationship between the two or the different modules

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/pom.xml]: You can use it test spark/flink with once code. Not only spark.

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/resources/clickhouse/fake_to_clickhouse.conf [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+source {
+  # This is a example source plugin **only for test and demonstrate the feature source plugin**
+  FakeSource {
+    result_table_name = "fake"
+    schema = {
+      fields {
+        name = "string"
+        age = "int"
+      }
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/resources/clickhouse/fake_to_clickhouse.conf]: test all datatypes?

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/pom.xml (hunk 1 above)
Comment (ashulin) [reply seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/pom.xml]: @FWLamb `seatunnel-{enigne}-connector-v2-e2e` modules are old implementation, which requires e2e testing for each engine (spark 3.3 and seatunnel engine will be added later), there will be a lot of duplicate code, which is boring and tiring.
`seatunnel-connector-v2-e2e`  unifies the e2e of the `connector-v2` and can support the e2e test of multiple engines at the same time.
The status quo is a legacy issue, I will upgrade to a new implementation as soon as possible.

File: seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/clickhouse/ClickhouseIT.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+
+
+    @BeforeEach
+    @SuppressWarnings("checkstyle:MagicNumber")
```
Comment (ashulin) [inline seatunnel-e2e/seatunnel-spark-connector-v2-e2e/connector-clickhouse-spark-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2/clickhouse/ClickhouseIT.java]:
```suggestion
```
For classes suffixed with IT or Test, it is automatically suppressed.

**Review by ashulin — COMMENTED** [review]
Please implement uniform E2E IT;
Note: Spark don‘t support the TIME type currently, so avoid testing it;
You can see #2924 or  #2946

**Pull request closed without merging**